vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/DetachReattachPage.java

```diff
@@ -68,7 +89,22 @@ public DetachReattachPage() {
                 });
         resetSortingButton.setId("reset-sorting-button");
 
+        Span errorMessage = new Span();
+        errorMessage.setId("error-message");
+
+        NativeButton btnSetCustomErrorHandler = new NativeButton(
+                "Set custom error handler", e -> {
+                    // Set error handler to show errors in the UI
+                    VaadinSession.getCurrent().setErrorHandler(event -> {
+                        errorMessage.setText(
+                                "Error: " + event.getThrowable().getMessage());
+                    });
+                });
+        btnSetCustomErrorHandler.setId("set-custom-error-handler-button");
```

Comment:
nit: Why the custom error handler instead of using `checkLogsForErrors();` in the test case?

Reply:
AFAIK `checkLogsForErrors()` only checks the client side logs. In this case, the error can only be verified from the server side and it doesn't show anything in the browser. That's why I had to add the handler there.

Reply:
For some reason, I wasn't seeing the errors before, but I can confirm that `checkForLogs` works to assert that the error is gone.

Reply:
Actually, I wonder if this only works when running in dev mode? Could it give false positives in production mode if the server-side error wouldn't get reflected on the client?

Reply:
I tried running it with `-Dvaadin.productionMode` as it seems to be used in some of the commands in the `build.sh` script and the test failed if I remove the change in Grid.